demos/greengrass/greengrass_demo_local_auth/greengrass_demo_local_auth.c

```diff
@@ -0,0 +1,922 @@
+    transport.pNetworkContext = pNetworkContext;
+    transport.send = Openssl_Send;
+    transport.recv = Openssl_Recv;
+
+    /* Fill the values for network buffer. */
+    networkBuffer.pBuffer = buffer;
+    networkBuffer.size = NETWORK_BUFFER_SIZE;
+
+    /* Initialize MQTT library. */
+    mqttStatus = MQTT_Init( pMqttContext,
+                            &transport,
+                            Clock_GetTimeMs,
+                            eventCallback,
+                            &networkBuffer );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        returnStatus = EXIT_FAILURE;
+        LogError( ( "MQTT init failed: Status = %s.", MQTT_Status_strerror( mqttStatus ) ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int subscribePublishLoop( MQTTContext_t * pMqttContext )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    uint32_t publishCount = 0;
+    const uint32_t maxPublishCount = MQTT_PUBLISH_COUNT_PER_LOOP;
+
+    assert( pMqttContext != NULL );
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* The client is now connected to the broker. Subscribe to the topic
+         * as specified in MQTT_EXAMPLE_TOPIC at the top of this file by sending a
+         * subscribe packet. This client will then publish to the same topic it
+         * subscribed to, so it will expect all the messages it sends to the broker
+         * to be sent back to it from the broker.  */
+        LogInfo( ( "Subscribing to the MQTT topic %.*s.",
+                   MQTT_EXAMPLE_TOPIC_LENGTH,
+                   MQTT_EXAMPLE_TOPIC ) );
+        returnStatus = subscribeToTopic( pMqttContext );
+    }
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Process incoming packet from the broker. Acknowledgment for subscription
+         * ( SUBACK ) will be received here. However after sending the subscribe, the
+         * client may receive a publish before it receives a subscribe ack. Since this
+         * demo is subscribing to the topic to which no one is publishing, probability
+         * of receiving publish message before subscribe ack is zero; but application
+         * must be ready to receive any packet. This demo uses MQTT_ProcessLoop to
+         * receive packet from network. */
+        mqttStatus = MQTT_ProcessLoop( pMqttContext );
+
+        if( mqttStatus != MQTTSuccess )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "MQTT_ProcessLoop returned with status = %s.",
+                        MQTT_Status_strerror( mqttStatus ) ) );
+        }
+    }
+
+    /* Check if recent subscription request has been rejected. globalSubAckStatus is updated
+     * in eventCallback to reflect the status of the SUBACK sent by the broker. */
+    if( ( returnStatus == EXIT_SUCCESS ) && ( globalSubAckStatus == MQTTSubAckFailure ) )
+    {
+        LogInfo( ( "Server rejected subscription request." ) );
```

Comment:
Nit: Should this be WARN level?

Reply:
Yeah, probably; it was Info, since other demos have it info